server/src/main/java/org/elasticsearch/index/mapper/BooleanScriptFieldType.java

```diff
@@ -37,22 +37,47 @@
         new Builder<>(name, BooleanFieldScript.CONTEXT, BooleanFieldScript.PARSE_FROM_SOURCE) {
             @Override
             RuntimeField newRuntimeField(BooleanFieldScript.Factory scriptFactory) {
-                return new BooleanScriptFieldType(name, scriptFactory, getScript(), meta(), this);
+                return runtimeField(name, this, scriptFactory, script.get(), meta.get());
             }
         });
 
-    public BooleanScriptFieldType(String name) {
-        this(name, BooleanFieldScript.PARSE_FROM_SOURCE, null, Collections.emptyMap(), (builder, params) -> builder);
+    private static RuntimeField runtimeField(
+        String name,
+        ToXContent toXContent,
+        BooleanFieldScript.Factory scriptFactory,
+        Script script,
+        Map<String, String> meta
+    ) {
+        return new ScriptRuntimeField(name, toXContent) {
+            @Override
+            public String typeName() {
+                return BooleanFieldMapper.CONTENT_TYPE;
+            }
+
+            @Override
+            public MappedFieldType asMappedFieldType() {
+                return new BooleanScriptFieldType(name, scriptFactory, script, meta);
+            }
+        };
+    }
+
+    public static RuntimeField sourceOnly(String name) {
```

Comment:
I am wondering if it makes sense to rework the structure of these fields, as now a parser uses a builder to parse a RuntimeField, which returns the field type. Does it all fit within the field type class? Would we do it this way if we were starting from scratch?

Reply:
It's analogous to Parser -> Builder -> FieldMapper -> MappedFieldType for standard fields.  We could I suppose pull the RuntimeField out into their own class files, but they're such small implementations that I didn't think it was worth it.